flink-libraries/flink-table/src/main/scala/org/apache/flink/api/scala/table/expressionDsl.scala

```diff
@@ -325,33 +337,42 @@ trait ImplicitExpressionOperations {
     */
   def day = toMilliInterval(expr, MILLIS_PER_DAY)
 
-    /**
+  /**
     * Creates an interval of the given number of hours.
     *
     * @return interval of milliseconds
     */
   def hour = toMilliInterval(expr, MILLIS_PER_HOUR)
 
-    /**
+  /**
     * Creates an interval of the given number of minutes.
     *
     * @return interval of milliseconds
     */
   def minute = toMilliInterval(expr, MILLIS_PER_MINUTE)
 
-    /**
+  /**
     * Creates an interval of the given number of seconds.
     *
     * @return interval of milliseconds
     */
   def second = toMilliInterval(expr, MILLIS_PER_SECOND)
 
-    /**
+  /**
     * Creates an interval of the given number of milliseconds.
     *
     * @return interval of milliseconds
     */
   def milli = toMilliInterval(expr, 1)
+
+  // row type
+
+  /**
+    * Creates a number defining an amount of rows.
+    *
+    * @return number of rows
+    */
+  def rows = expr
```

Comment:
I'm still not sure if we need an additional data type for that. Actually we could also just allow `Int` and `Long` and remove the `rows` completely. It is just syntactic sugar without additional advantage. It might also be good if we would allow constant arithmetic in future, such as `10*10.rows`. But if you think a row interval makes sense, I will add it.